File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        
        int maxSumTillHere=0;
        int ans=Integer.MIN_VALUE;
        
        
        for(int i=0;i<nums.length;i++)
        {
            maxSumTillHere+=nums[i];
            
            ans=Math.max(ans,maxSumTillHere);
            maxSumTillHere=Math.max(0,maxSumTillHere);
        }
        
        return ans;
        
    }
}

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        
        
        int max_till_here=0;
        int ans=INT_MIN;
        
        for(int i=0;i<nums.size();i++)
        {
            max_till_here+=nums[i];
            ans=max(ans,max_till_here);
            
            max_till_here=max(0,max_till_here);
            
        }
        return ans;
    }
};